### ISSUE TITLE
Weak Type Detection - not being detected with `const`

### ISSUE THREAD

**Issue by alan-agius4** [Microsoft/TypeScript#18250]
**TypeScript Version:**  2.5.2

**Code**

```ts

function foo(_opts: Options) {
    // do something;
    return;
}

interface Options {
    data?: any;
    show?: boolean;
}

interface OptionsExtended extends Options {
    title?: string;
}

const opts = {
    show: true,
    title: "title"
};

// this works
foo(opts);

// this doesn't
foo({
    show: true,
    title: "title"
});
```

**Expected behavior:**
Both of them to emit an error

**Actual behavior:**
the first one works and doesn't emit an error

**Comment by kitsonk** [comment]
This doesn't have to do with weak type detection.  This has to do with excess property checks between _fresh_ object literals and inferred types.  I just explained it [here](https://github.com/Microsoft/TypeScript/issues/18233#issuecomment-326924398).
